WalletWasabi/Rpc/JsonRpcResponse.cs

```diff
@@ -15,51 +15,54 @@ public class JsonRpcResponse
 		[JsonRpcErrorCodes.InternalError] = "Internal error",
 	};
 
-	private JsonRpcResponse(string id, object? result, object? error)
+	protected JsonRpcResponse(string id)
 	{
 		Id = id;
-		Result = result;
-		Error = error;
 	}
 
 	[JsonProperty("jsonrpc", Order = 0)]
 	public string JsonRpc => "2.0";
 
-	[JsonProperty("result", Order = 1)]
-	public object? Result { get; }
-
-	[JsonProperty("error", Order = 1)]
-	public object? Error { get; }
-
 	[JsonProperty("id", Order = 3)]
 	public string Id { get; }
 
-	public static JsonRpcResponse CreateResultResponse(string id, object result)
-	{
-		return new JsonRpcResponse(id, result, null);
-	}
+	public static JsonRpcSuccessResponse CreateResultResponse(string id, object? result = null) =>
+		new(id, result);
 
-	public static JsonRpcResponse CreateErrorResponse(string id, JsonRpcErrorCodes code, string? message = null)
-	{
-		var error = new
-		{
-			code,
-			message = message ?? GetMessage(code)
-		};
-		return new JsonRpcResponse(id, null, error);
-	}
+	public static JsonRpcErrorResponse CreateErrorResponse(string id, JsonRpcErrorCodes code, string? customMessage = null) =>
+		new(id, code, customMessage ?? GetDefaultMessageFor(code));
+
+	public string ToJson(JsonSerializerSettings serializerSettings) =>
+		JsonConvert.SerializeObject(this, serializerSettings);
 
-	private static string GetMessage(JsonRpcErrorCodes code)
+	private static string GetDefaultMessageFor(JsonRpcErrorCodes code) =>
+		Messages.TryGetValue(code, out var rpcErrorMessage)
+			? rpcErrorMessage
+			: "Server error";
+}
+
+public record JsonRpcSuccessResponse : JsonRpcResponse
+{
+	public JsonRpcSuccessResponse(string id, object? result)
+		: base(id)
 	{
-		if (Messages.TryGetValue(code, out var message))
-		{
-			return message;
-		}
-		return "Server error";
+		Result = result;
 	}
 
-	public string ToJson(JsonSerializerSettings serializerSettings)
+	[JsonProperty("result", Order = 1)]
+	public object? Result { get; }
+}
+
+public record JsonRpcErrorResponse : JsonRpcResponse
+{
+	public JsonRpcErrorResponse(string id, JsonRpcErrorCodes code, string message)
+		: base(id)
 	{
-		return JsonConvert.SerializeObject(this, serializerSettings);
+		Error = new ErrorObject(code, message);
 	}
+
+	[JsonProperty("error", Order = 1)]
+	public ErrorObject Error { get; }
+
+	public record ErrorObject(JsonRpcErrorCodes code, string message);
```

Comment:
```suggestion
	public record ErrorObject(JsonRpcErrorCodes Code, string Message);
```
[IDE1006](https://learn.microsoft.com/en-us/dotnet/fundamentals/code-analysis/style-rules/naming-rules)

Reply:
- Or suppress the violation with pragma warning disable.

But that's just a bandage, and this error is only in visual studio (but it still builds).
I leave the decision up to you, I'd go with the 1st or 2nd suggestion.